aws/resource_aws_glue_trigger.go

```diff
@@ -423,9 +463,14 @@ func flattenGlueActions(actions []*glue.Action) []interface{} {
 	for _, action := range actions {
 		m := map[string]interface{}{
 			"arguments": aws.StringValueMap(action.Arguments),
-			"job_name":  aws.StringValue(action.JobName),
 			"timeout":   int(aws.Int64Value(action.Timeout)),
 		}
+
+		if action.JobName != nil && len(*action.JobName) > 0 {
+			m["job_name"] = aws.StringValue(action.JobName)
+		} else if action.CrawlerName != nil && len(*action.CrawlerName) > 0 {
+			m["crawler_name"] = aws.StringValue(action.CrawlerName)
+		}
```

Comment:
This can be simplified since `aws.StringValue()` automatically converts `nil` to `""` (also without assuming any precedence):

```go
if v := aws.StringValue(action.CrawlerName); v != "" {
  m["crawler_name"] = v
}

if v := aws.StringValue(action.JobName); v != "" {
  m["job_name"] = v
}
```